pkgs/tools/networking/ntopng/default.nix

```diff
@@ -1,41 +1,44 @@
-{ stdenv, fetchurl, libpcap,/* gnutls, libgcrypt,*/ libxml2, glib
-, geoip, geolite-legacy, sqlite, which, autoreconfHook, git
-, pkgconfig, groff, curl, json_c, luajit, zeromq, rrdtool
+{ stdenv, fetchurl, fetchFromGitHub, libpcap,/* gnutls, libgcrypt,*/ libxml2, glib
+, geoip, geolite-legacy, sqlite, which, autoreconfHook, git, mariadb, readline80
+, pkgconfig, groff, curl, json_c, luajit, zeromq, rrdtool, libmaxminddb
 }:
 
 # ntopng includes LuaJIT, mongoose, rrdtool and zeromq in its third-party/
 # directory, but we use luajit, zeromq, and rrdtool from nixpkgs
 
-stdenv.mkDerivation rec {
-  name = "ntopng-2.0";
+let
+  ndpi = fetchurl {
+    url = "https://github.com/ntop/nDPI/archive/2.6.tar.gz";
```

Comment:
GitHub-generated _archives_ are not stable over time. You should rather use `fetchzip` or `fetchFromGitHub` here.

Reply:
Too bad, but what do I do then? I mean, after fetching the files, say via `fetchFromGitHub`, I need to put it into the build directory, since it needs to be writable. Do I have to make it a (sub-)derivation then?

Reply:
After some back-and-forth, I managed to change it to use `fetchFromGitHub` instead.